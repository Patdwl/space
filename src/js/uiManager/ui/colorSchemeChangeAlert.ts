import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { ColorRuleSet } from '@app/js/colorManager/colorSchemeManager';
import { toast } from './toast';

export const colorSchemeChangeAlert = (newScheme: ColorRuleSet) => {
  const { uiManager } = keepTrackApi.programs;
  // Don't Make an alert the first time!
  if (typeof uiManager.lastColorScheme == 'undefined') {
    uiManager.lastColorScheme = newScheme;
    return;
  }

  const { colorSchemeManager } = keepTrackApi.programs;

  // Don't make an alert unless something has really changed
  if (uiManager.lastColorScheme == newScheme) return;

  for (const scheme in colorSchemeManager) {
    if (newScheme == colorSchemeManager[scheme] && scheme !== 'currentColorScheme' && scheme !== 'lastColorScheme') {
      // record the new color scheme
      uiManager.lastColorScheme = newScheme;
      // Make an alert
      switch (scheme) {
        case 'default':
        case 'group':
          toast(`Color Scheme Changed to Object Types`, 'normal', false);
          break;
        case 'velocity':
          toast(`Color Scheme Changed to Velocity`, 'normal', false);
          break;
        case 'sunlight':
          toast(`Color Scheme Changed to Sunlight`, 'normal', false);
          break;
        case 'countries':
        case 'groupCountries':
          toast(`Color Scheme Changed to Countries`, 'normal', false);
          break;
        case 'leo':
          toast(`Color Scheme Changed to Near Earth`, 'normal', false);
          break;
        case 'geo':
          toast(`Color Scheme Changed to Deep Space`, 'normal', false);
          break;
        case 'ageOfElset':
          toast(`Color Scheme Changed to Elset Age`, 'normal', false);
          break;
        case 'rcs':
          toast(`Color Scheme Changed to Radar Cross Section`, 'normal', false);
          break;
        case 'smallsats':
          toast(`Color Scheme Changed to Small Satellites`, 'normal', false);
          break;
        case 'lostobjects':
          toast(`Color Scheme Changed to Lost Objects`, 'normal', false);
          break;
        default:
          toast(`Color Scheme Changed to ${scheme}`, 'normal', false);
          console.debug(`${scheme} missing from alert list!`);
          break;
      }
    }
  }
};
