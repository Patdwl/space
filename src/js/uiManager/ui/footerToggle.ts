import { getEl } from '@app/js/lib/helpers';

let isFooterShown = true;
export const footerToggle = function () {
  if (isFooterShown) {
    isFooterShown = false;
    getEl('sat-infobox')?.classList.add('sat-infobox-fullsize');
    getEl('nav-footer')?.classList.add('footer-slide-trans');
    getEl('nav-footer')?.classList.remove('footer-slide-up');
    getEl('nav-footer')?.classList.add('footer-slide-down');
    getEl('nav-footer-toggle').innerHTML = '&#x25B2;';
  } else {
    isFooterShown = true;
    getEl('sat-infobox')?.classList.remove('sat-infobox-fullsize');
    getEl('nav-footer')?.classList.add('footer-slide-trans');
    getEl('nav-footer')?.classList.remove('footer-slide-down');
    getEl('nav-footer')?.classList.add('footer-slide-up');
    getEl('nav-footer-toggle').innerHTML = '&#x25BC;';
  }
  // After 1 second the transition should be complete so lets stop moving slowly
  setTimeout(() => {
    getEl('nav-footer')?.classList.remove('footer-slide-trans');
  }, 1000);
};
