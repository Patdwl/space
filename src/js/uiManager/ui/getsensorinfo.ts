import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { SensorObject } from '@app/js/api/keepTrackTypes';
import { getEl } from '@app/js/lib/helpers';

export const getsensorinfo = () => {
  const { currentSensor }: { currentSensor: SensorObject[] } = keepTrackApi.programs.sensorManager;

  const firstSensor = currentSensor[0];
  getEl('sensor-latitude').innerHTML = firstSensor.lat.toString();
  getEl('sensor-longitude').innerHTML = firstSensor.lon.toString();
  getEl('sensor-minazimuth').innerHTML = firstSensor.obsminaz.toString();
  getEl('sensor-maxazimuth').innerHTML = firstSensor.obsmaxaz.toString();
  getEl('sensor-minelevation').innerHTML = firstSensor.obsminel.toString();
  getEl('sensor-maxelevation').innerHTML = firstSensor.obsmaxel.toString();
  getEl('sensor-minrange').innerHTML = firstSensor.obsminrange.toString();
  getEl('sensor-maxrange').innerHTML = firstSensor.obsmaxrange.toString();
};
