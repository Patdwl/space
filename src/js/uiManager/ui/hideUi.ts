import { getEl } from '@app/js/lib/helpers';
import { uiManager } from '../uiManager';

export const hideUi = () => {
  if (uiManager.isUiVisible) {
    getEl('header').style.display = 'none';
    getEl('ui-wrapper').style.display = 'none';
    getEl('nav-footer').style.display = 'none';
    uiManager.isUiVisible = false;
  } else {
    getEl('header').style.display = 'block';
    getEl('ui-wrapper').style.display = 'block';
    getEl('nav-footer').style.display = 'block';
    uiManager.isUiVisible = true;
  }
};
