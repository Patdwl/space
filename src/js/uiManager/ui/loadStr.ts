import { getEl } from '@app/js/lib/helpers';

export const loadStr = (str: string) => {
  if (str == '') {
    getEl('loader-text').innerHTML = '';
    return;
  }
  if (str == 'math') {
    getEl('loader-text').innerHTML = 'Attempting to Math...';
  }

  switch (str) {
    case 'science':
      getEl('loader-text').innerHTML = 'Locating Science...';
      break;
    case 'science2':
      getEl('loader-text').innerHTML = 'Found Science...';
      break;
    case 'dots':
      getEl('loader-text').innerHTML = 'Drawing Dots in Space...';
      break;
    case 'satIntel':
      getEl('loader-text').innerHTML = 'Integrating Satellite Intel...';
      break;
    case 'radarData':
      getEl('loader-text').innerHTML = 'Importing Radar Data...';
      break;
    case 'painting':
      getEl('loader-text').innerHTML = 'Painting the Earth...';
      break;
    case 'coloring':
      getEl('loader-text').innerHTML = 'Coloring Inside the Lines...';
      break;
    case 'elsets':
      getEl('loader-text').innerHTML = 'Locating ELSETs...';
      break;
    case 'models':
      getEl('loader-text').innerHTML = 'Loading 3D Models...';
      break;
    case 'easterEgg':
      getEl('loader-text').innerHTML = 'Llama Llama Llama Duck!';
  }
};
