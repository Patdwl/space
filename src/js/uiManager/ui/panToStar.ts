import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { SatObject } from '@app/js/api/keepTrackTypes';

export const panToStar = (c: SatObject): void => {
  const { objectManager, satSet, lineManager, mainCamera, starManager } = keepTrackApi.programs;

  // Try with the pname
  let satId = satSet.getIdFromStarName(c.name);
  let sat = satSet.getSat(satId);

  if (sat == null) throw new Error('Star not found');

  lineManager.clear();
  if (objectManager.isStarManagerLoaded) {
    starManager.isAllConstellationVisible = false;
  }

  lineManager.create('ref', [sat.position.x, sat.position.y, sat.position.z], [1, 0.4, 0, 1]);
  mainCamera.cameraType.current = mainCamera.cameraType.Offset;
  mainCamera.lookAtObject(sat, false);
};
