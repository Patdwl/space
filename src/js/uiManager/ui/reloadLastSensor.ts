import { SensorObject } from '@app/js/api/keepTrackTypes';
import { getsensorinfo } from './getsensorinfo';
import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { legendMenuChange } from '..';

export const reloadLastSensor = () => {
  let currentSensor: SensorObject;
  try {
    currentSensor = JSON.parse(localStorage.getItem('currentSensor'));
  } catch (e) {
    currentSensor = null;
  }
  // istanbul ignore next
  if (currentSensor !== null) {
    try {
      const { sensorManager, mainCamera, timeManager } = keepTrackApi.programs;

      // If there is a staticnum set use that
      if (typeof currentSensor[0] == 'undefined' || currentSensor[0] == null) {
        sensorManager.setSensor(null, currentSensor[1]);
        getsensorinfo();
        legendMenuChange('default');
      } else {
        // If the sensor is a string, load that collection of sensors
        if (typeof currentSensor[0].shortName == 'undefined') {
          sensorManager.setSensor(currentSensor[0], currentSensor[1]);
          getsensorinfo();
          legendMenuChange('default');
          mainCamera.lookAtLatLon(sensorManager.selectedSensor.lat, sensorManager.selectedSensor.lon, sensorManager.selectedSensor.zoom, timeManager.selectedDate);
        } else {
          // Seems to be a single sensor without a staticnum, load that
          sensorManager.setSensor(sensorManager.sensorList[currentSensor[0].shortName], currentSensor[1]);
          getsensorinfo();
          legendMenuChange('default');
          mainCamera.lookAtLatLon(sensorManager.selectedSensor.lat, sensorManager.selectedSensor.lon, sensorManager.selectedSensor.zoom, timeManager.selectedDate);
        }
      }
    } catch (e) {
      // Clear old settings because they seem corrupted
      try {
        localStorage.setItem('currentSensor', null);
        console.warn('Saved Sensor Information Invalid');
      } catch {
        // do nothing
      }
    }
  }
};
