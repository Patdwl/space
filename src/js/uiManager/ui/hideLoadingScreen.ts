import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { getEl } from '@app/js/lib/helpers';
import { mobileManager } from '@app/js/uiManager/mobile/mobileManager';
import { loadStr } from './loadStr';

export const hideLoadingScreen = () => {
  const { drawManager } = keepTrackApi.programs;
  // Don't wait if we are running Jest
  if (drawManager.sceneManager.earth.isUseHiRes && drawManager.sceneManager.earth.isHiResReady !== true) {
    setTimeout(function () {
      hideLoadingScreen();
    }, 100);
    return;
  }

  // Display content when loading is complete.
  getEl('canvas-holder').style.display = 'block';

  mobileManager.checkMobileMode();

  if (settingsManager.isMobileModeEnabled) {
    loadStr('math');
    getEl('loading-screen').style.display = 'none';
    getEl('loading-earth').style.display = 'none';
  } else {
    // Loading Screen Resized and Hidden
    setTimeout(function () {
      getEl('loading-screen').classList.remove('full-loader');
      getEl('loading-screen').classList.add('mini-loader-container');
      getEl('logo-inner-container').classList.add('mini-loader');
      getEl('logo-text').innerHTML = '';
      getEl('logo-text-version').innerHTML = '';
      getEl('loading-screen').style.display = 'none';
      loadStr('math');
    }, 100);
  }
};
