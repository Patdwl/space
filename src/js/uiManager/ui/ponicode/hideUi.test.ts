import { keepTrackApiStubs } from '@app/js/api/apiMocks';
import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { KeepTrackPrograms } from '@app/js/api/keepTrackTypes';
import * as hideUi from '@app/js/uiManager/ui/hideUi';

keepTrackApi.programs = <KeepTrackPrograms>(<unknown>{ ...keepTrackApi.programs, ...keepTrackApiStubs.programs });

// @ponicode
describe('hideUi.hideUi', () => {
  test('0', () => {
    let result: any = hideUi.hideUi();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = () => {
      hideUi.hideUi();
      hideUi.hideUi();
      hideUi.hideUi();
    };
    expect(result()).toMatchSnapshot();
  });
});
