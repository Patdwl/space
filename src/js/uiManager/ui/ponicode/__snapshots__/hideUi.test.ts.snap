// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hideUi.hideUi 0 1`] = `undefined`;

exports[`hideUi.hideUi 1 1`] = `undefined`;
